client/app/api/course/MaterialFolders.js

```diff
@@ -0,0 +1,26 @@
+import BaseCourseAPI from './Base';
+
+export default class MaterialFoldersAPI extends BaseCourseAPI {
+  /**
+  * Upload files to the specified folder.
+  *
+  * @param {number} folderId
+  * @param {array} files - A list of files from file input.
+  * @return {Promise}
+  * success response: { materials: [{:id, :name, :url, :updated_at}] } - A list of materials that
```

Comment:
I didn't find any way to document `Promise` in JSDoc. So I'd just describe the server responses using text. 

The purpose here is for us to know the format of response just by reading the client code ( without the need of debugging on the server side ), I believe this will improve the readability of the client code. 

it will be very useful for you to understand the code like:
```javascript
.then(response => {
    this.setState(response.data); // You pronbably have no clue what's the state now.
})
```

Let me know your suggestions on the documentation formats, and let's follow the convention when dealing with server API next time.